feat(docs): First pass at Gatsby Themes documentation

johno opened the pull request:
## Description

This is the first pass at some preliminary documentation on themes. Would love any and all feedback!

These pages are intentionally not linked to in the side bar for now, but this will give us something to link to as we continue to refine the docs and add more content.

### TODO:

- [ ] Create example starter that uses a theme for getting started page

marcysutton commented on docs/docs/themes/introduction.md:
```diff
@@ -0,0 +1,54 @@
+
+Theming in Gatsby means you can now avoid the boilerplate required when setting up a new project that’s commonplace in the community. That means we can arrive at a shared convention for something like a blog, and not be required to write GraphQL or configure a plugin unless you want to. No longer do you have to reinvent the wheel for each new Gatsby site you want to build.
+
+It provides a foundation to build upon, and since themes can inherit from other themes you can you can focus on building out new abstractions or functionality.
```
```suggestion
It provides a foundation to build upon, and since themes can inherit from other themes you can you can focus on building out new abstractions or functionality. And by including [accessibility](./docs/themes/building-themes#accessibility/) in your theme, you can contribute back to create a more accessible web!
```
Not sure if that would be the correct link for the section I proposed above, but you get the idea!

johno commented on docs/docs/themes/api-reference.md:
```diff
@@ -0,0 +1,126 @@
+## Component Shadowing
+
+Gatsby Themes allow you to customize any file in a theme's `src` directory by following a file naming convention.
+If you're using `gatsby-theme-tomato` which uses a `ProfileCard` component located at `src/components/ProfileCard.js` you can override the component by creating `src/gatsby-theme-tomato/components/profile-card.js`. If you want to see what props are passed you can do so by putting the props into a `pre` tag:
```
Good point, yeah we should probably use kebab case for file names since that's Gatsby convention. I'll run through and make sure that it's reflected everywhere.

johno commented on docs/docs/themes/getting-started.md:
````diff
@@ -0,0 +1,51 @@
+npx gatsby new my-blog https://github.com/gatsbyjs/gatsby-starter-blog-theme
+```
+
+With theme-based starters, you will see that you're initially presented with a lighter weight `gatsby-config.js`. The goal with most themes is to allow you to get up and running with something like a blog in seconds. Now you can focus on writing and gradually introduce changes and customizations only if you want to.
````
Reworded to "When you use a starter that's built with a theme". Maybe that's a better way to phrase it?

johno commented on docs/docs/themes/customizing-a-theme.md:
````diff
@@ -0,0 +1,38 @@
+}
+```
+
+## Component Shadowing
````
I've gone ahead and combined the `customizing-a-theme` page with the `api-reference` page since there was a lot of overlap.

johno commented:
Thank you for the great review @marcysutton. I've addressed your feedback and reworked a few things. Please let me know if you'd like to see other changes.

DSchau commented on docs/docs/themes/api-reference.md:
```diff
@@ -0,0 +1,126 @@
+title: Themes API Reference
+---
+
+> ⚠⚠ Gatsby Themes are currently experimental ⚠⚠
```
We need `mdx` 😆 

```
import Alert from 'components/alert'

<Alert type="warning">Gatsby Themes are currently experimental</Alert>
```

marcysutton commented:
Woooo! Exciting!
